Contest_API.md

```diff
@@ -1448,23 +1475,22 @@ For some common award cases the following IDs should be used.
 | group-winner-\<id>        | Current leader(s) in group \<id>. Empty if no team has scored.                                                             | Winner(s) of group \<id>.                 |
 | organization-winner-\<id> | Current leader(s) of organization \<id>. Empty if no team has scored.                                                      | Winner(s) of organization \<id>.          | Not useful in contest with only one team per organization (e.g. the WF).
 
-#### POST, PATCH, and DELETE awards
+#### POST, PUT, PATCH, and DELETE awards
 
-Clients with the `admin` role may make changes to awards using the normal [HTTP methods](#http-methods) as specified above. Specifically, they can POST new awards, PATCH one or more attributes, or DELETE an existing award. All requests must include the id attribute.
+Clients with the `admin` role may make changes to awards using the normal [HTTP methods](#http-methods) as specified above. Specifically, they can POST new awards, replace one with PUT, PATCH one or more attributes, or DELETE an existing award. POSTs should include the `id` when creating new awards of the known award types above; if the id is omitted the server will assign one.
```

Comment:
Does it make sense for the server to generate an `id` if none is given? I'd find it more natural that the server is allowed to reject a POST without an `id`. The same might also apply to POSTs to problems and languages (although we don't support that yet).

Reply:
I was assuming the id of a random/unknown award didn't matter, so a generated one was just as good as the client picking one. Are you totally against generated award ids, or just best practice to include one?

Reply:
Even more natural would be to `PUT` if you must specify the `id`.